### PR TITLE
Add User entity

### PR DESCRIPTION
**Pull request by AntonBabychP1T**
Add userDto
Add AuthenticationController with register method
Add FieldMatch annotation
Add RegistrationException and using in the GlobalExceptionHandler

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -71,6 +71,13 @@
             <groupId>org.hibernate.validator</groupId>
             <artifactId>hibernate-validator</artifactId>
         </dependency>
+        <!-- https://mvnrepository.com/artifact/commons-beanutils/commons-beanutils -->
+        <dependency>
+            <groupId>commons-beanutils</groupId>
+            <artifactId>commons-beanutils</artifactId>
+            <version>1.9.4</version>
+        </dependency>
```
Comment (sokimaaa) [inline pom.xml]: Why do you need this?
```suggestion
```

File: src/main/java/store/bookstoreapp/exception/RegistrationException.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package store.bookstoreapp.exception;
+
+public class RegistrationException extends Exception {
```
Comment (sokimaaa) [inline src/main/java/store/bookstoreapp/exception/RegistrationException.java]: prefer using unchecked exceptions over checked

File: src/main/java/store/bookstoreapp/service/UserService.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package store.bookstoreapp.service;
+
+import store.bookstoreapp.dto.user.UserRegistrationRequestDto;
+import store.bookstoreapp.dto.user.UserResponseDto;
+import store.bookstoreapp.exception.RegistrationException;
+
+public interface UserService {
+    UserResponseDto register(UserRegistrationRequestDto requestDto) throws RegistrationException;
```
Comment (sokimaaa) [inline src/main/java/store/bookstoreapp/service/UserService.java]:
```suggestion
    UserResponseDto register(UserRegistrationRequestDto requestDto);
```

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 4]
```diff
@@ -1,3 +1,5 @@
 databaseChangeLog:
   - include:
       file: db/changelog/changes/01-create-books-table.yaml
+  - include:
+      file: db/changelog/changes/02-create-users-table.yaml
```
Comment (sokimaaa) [inline src/main/resources/db/changelog/db.changelog-master.yaml:5]: consider adding `insert users` script

File: src/main/java/store/bookstoreapp/controller/AuthenticationController.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import store.bookstoreapp.exception.RegistrationException;
+import store.bookstoreapp.service.UserService;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/api/auth")
+public class AuthenticationController {
+    private final UserService userService;
+
+    @PostMapping
```
Comment (sokimaaa) [inline src/main/java/store/bookstoreapp/controller/AuthenticationController.java]:
```suggestion
    @PostMapping("register")
```

File: src/main/java/store/bookstoreapp/validation/FieldMatchValidator.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        try {
+            Field firstField = value.getClass().getDeclaredField(firstFieldName);
+            firstField.setAccessible(true);
+            final Object firstValue = firstField.get(value);
+
+            Field secondField = value.getClass().getDeclaredField(secondFieldName);
+            secondField.setAccessible(true);
+            final Object secondValue = secondField.get(value);
+
+            return firstValue != null && firstValue.equals(secondValue);
```
Comment (Rommelua) [inline src/main/java/store/bookstoreapp/validation/FieldMatchValidator.java]: return  Objects.equals()

**Review by Rommelua — APPROVED** [review]

**Pull request merged**